packages/svelte/src/internal/client/hydration.js

```diff
@@ -68,12 +68,11 @@ export function hydrate_block_anchor(anchor_node, is_controlled) {
 			let fragment = target_node.$$fragment;
 			if (fragment === undefined) {
 				fragment = get_hydration_fragment(target_node);
-				// @ts-ignore remove to prevent memory leaks
-				target_node.$$fragment = undefined;
 			}
```

Comment:
Because we call `hydrate_block_anchor` on the same anchor many times. If we remove it here, then it will break when it gets used again. This happens when hydrating components. i.e.

```svelte
<div>
  <Component>
    <Component2 />
  </Component
</div>
```

They all have the same comment node from within the `<div>`

Reply:
Done, using another approach